woocommerce/woocommerce-android
# [Product discount] Manual product discount flow

## Manual product discount flow

This PR adds the possibility to apply a discount directly to an item in order when creating or editing an order.

Closes: #9352


### Description


We're using the order update endpoint and modifying `Order.Item.total` according to the discount amount provided by the user. More details about discounting using order update endpoint: p1655830808613139-slack-CGPNUU63E

### Testing instructions


1. Create or edit an order containing the product(s) 
2. Click on an item and then "+ Add discount"
3. Enter a discount and return to the order edition screen
4. Verify order total is updated correctly and includes the discount amount

Focus on testing various scenarios including:
1. Multiple discounts 
2. Invalid discount amounts
3. Modifying the quantity of the discounted item
4. Try to find unhandled edge cases

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/4527432/2b791791-6fe5-4aca-877c-9ba30b956220

### Roadmap
In the next PRs, we'll add:
1. Percentage discount picker
2. Showing discount amount values in the order create/edit screen on the items list
5. Improving the UI of the product details screen
6. Analytics events
